src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -83,23 +83,26 @@
  */
 public class EntityCitizen extends EntityAgeable implements INpc
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 66 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') class "EntityCitizen" has 103 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)